barretenberg/cpp/src/barretenberg/vm2/simulation/merkle_check.test.cpp

```diff
@@ -1,24 +1,39 @@
+#include "barretenberg/vm2/simulation/merkle_check.hpp"
+
 #include <gmock/gmock.h>
 #include <gtest/gtest.h>
 #include <optional>
 
 #include "barretenberg/vm2/simulation/events/event_emitter.hpp"
 #include "barretenberg/vm2/simulation/events/merkle_check_event.hpp"
 #include "barretenberg/vm2/simulation/lib/merkle.hpp"
-#include "barretenberg/vm2/simulation/merkle_check.hpp"
+#include "barretenberg/vm2/simulation/testing/mock_execution_id_manager.hpp"
+#include "barretenberg/vm2/simulation/testing/mock_gt.hpp"
 #include "barretenberg/vm2/testing/macros.hpp"
 
 namespace bb::avm2::simulation {
 namespace {
 
 using testing::ElementsAre;
+using ::testing::NiceMock;
+
+class MerkleCheckSimulationTest : public ::testing::Test {
+  protected:
+    MerkleCheckSimulationTest() = default;
 
-TEST(MerkleCheckSimulationTest, AssertMembership)
-{
     EventEmitter<Poseidon2HashEvent> hash_emitter;
-    EventEmitter<Poseidon2PermutationEvent> perm_emitter;
-    Poseidon2 poseidon2(hash_emitter, perm_emitter);
+    // unused hence NoopEventEmitter
+    NoopEventEmitter<Poseidon2PermutationEvent> perm_emitter;
+    NoopEventEmitter<Poseidon2PermutationMemoryEvent> perm_mem_event_emitter;
+
+    NiceMock<MockExecutionIdManager> mock_execution_id_manager;
+    NiceMock<MockGreaterThan> mock_gt;
+    Poseidon2 poseidon2 =
+        Poseidon2(mock_execution_id_manager, mock_gt, hash_emitter, perm_emitter, perm_mem_event_emitter);
```

Comment:
Why do we use the real poseidon (which forces you to add all deps) and not just a StrictMock for it? You can even either
(1) mock the implementation to use the bb poseidon or,
(2) create a FakePoseidon2 which implements everything the way fast simulation would, and use that

Reply:
i think mainly because these tests are also checking that the correct number of events are being emitted by the poseidon2 trace.

I guess an alternative way could be to Mock the poseidon2 and `EXPECT(poseidon2, emit).Times(X)`? I didnt want to change the behavior of Alvaro's test though